portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,21 +28,52 @@
 import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private final class Comment {
+    private final String user;
+    private final String message;
+
+    public Comment(String user, String message) {
+      this.user = user;
+      this.message = message;
+    }
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> list = new ArrayList<>();
-    list.add("first message");
-    list.add("second message");
-    list.add("third message");
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Comment> comments = new ArrayList();
 
-    Gson gson = new Gson();
-    String json = gson.toJson(list);
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("title");
+      String user = (String) entity.getProperty("description");
+      Comment c = new Comment(user, comment);
+      comments.add(c);
```

Comment:
Uber nit.
Consider inlining new comment:
```suggestion
      comments.add(new Comment(user, comment));
```
The local variable is not really used and the suggestion above is also not very long in terms of line length.